### PR TITLE
Topic/website core docs

### PR DESCRIPTION
**Pull request by hippee-lee**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

There is no documentation for core components

Issue Number: N/A

## What is the new behavior?
This PR adds baseline content for core web-components. There were multiple contributors, on a long running branch. One commit per contributor was preserved except for smaller changes from me that came after. 

# Highlights

1. Added web-components documentation pages: overview/usage, code & API
2. Overview / usage content is hand curated and implemented closely to the design vision delivered via Figma
3. Code content is generated by parsing `*.story.ts` files to pull out code that is annotated with `@website` and then integrated with the content through Vue components - see the build directory under the website for scripts and details
4. Component API is generated with a wca script that generates json data about the public surface of the Core component API - data is consumed with a Vue component
5. get started for angular, web-components and icons have been moved to more natural locations and edited for v5
6 Website has pre-build and pre-start tasks that run the parsing scripts for code data

# Known Issues
1. Formatting for code isn't quite right
2. Web component API tables are incomplete - needs to be fixed by enhancing the JSDoc annotations for core components
3. Code tabs do not have a table-of-contents. The headers are generated based on whatever is in the parsed story data but the TOC is generated from markdown which we don't know at time of build, only after the data is pulled in.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by coryrylan — APPROVED** [review]

**Comment by coryrylan** [comment]
Adding an approval so we can merge to make it easier for the other active branches being worked on

File: apps/website/.vuepress/public/core-usage-demos/input-group/control-actions.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<form>
```
Comment (gnomeontherun) [inline apps/website/.vuepress/public/core-usage-demos/input-group/control-actions.html:1]: The reason these are in `public` is back to when we would dynamically load these files, they don't have to be here anymore.

File: apps/website/.vuepress/public/core-usage-demos/toggle/basic.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<cds-toggle>
+  <label>toggle</label>
+  <input type="checkbox" checked />
+</cds-toggle>
```
Comment (gnomeontherun) [inline apps/website/.vuepress/public/core-usage-demos/toggle/basic.html:4]: Why do we need Clarity City loaded here manually?

**Review by gnomeontherun — APPROVED** [review]

File: apps/website/.vuepress/public/core-usage-demos/toggle/basic.html (hunk 2 above)
Comment (hippee-lee) [reply apps/website/.vuepress/public/core-usage-demos/toggle/basic.html:4]: @mathisscott noticed that even though the core styles were loading, the font wasn't. This is the work around to get it loaded for now.

**Pull request merged**